src/main.rs

```diff
@@ -1,3 +1,269 @@
-fn main() {
-    println!("Hello, world!");
+use macroquad::prelude::*;
+//SET THE GRID VALUES
+const GRID_WIDTH: usize = 100;
+const GRID_HEIGHT: usize = 100;
+
+#[macroquad::main("BasicShapes")]
+async fn main() {
+    let mut matrix = vec![vec![0; GRID_WIDTH]; GRID_HEIGHT];
+
+    let mut is_running = false;
+    let mut is_grid_showing = false;
+    loop {
+        if is_key_pressed(KeyCode::Space) {
+            is_running = !is_running;
+        }
+        let status: &str;
+        if is_running {
+            status = "RUNNING"
+        } else {
+            status = "PAUSED"
+        }
+        if is_key_pressed(KeyCode::G) {
+            is_grid_showing = !is_grid_showing;
+        }
+        if !is_running && is_mouse_button_pressed(MouseButton::Left) {
+            process_clicks(&mut matrix);
+        }
+        //Color background
+        clear_background(WHITE);
+        //Draw the current state of the grid
+        draw(&matrix, &is_grid_showing);
+        //DRAW text
+        draw_text(status, 5.0, 30.0, 50.0, DARKBLUE);
+        draw_text(
+            "Cells can be edited on pause",
```

Comment:
If cells can only be edited while the game is paused then I think this message should be something like:

```suggestion
            "Cells must be edited on pause",
```

or:

```suggestion
            "Cells can only be edited on pause",
```

Reply:
Good suggestion, I liked the second option better. Done